posts/hands-on-guide-to-migrating-postgres-with-pglogical-and-aws-dms.md

```diff
@@ -0,0 +1,440 @@
+
+![Timeline Overview](../images/2023/07/migration_blog_diagrams_timeline.jpg)
+
+### Environment Notes
+
+Environment: For illustration purposes, consider this setup. A Client has a Primary Database, a Read Replica, and then a second "maintenance" Read Replica. Both read replicas are replicating from the primary database using built-in Postgres 9.6 "log shipping". See Postgres Replication - Log Shipping. Most of this plan can be adapted to work with different environment setups, I will try to describe in as much detail as possible where trade offs exist.
+
+Constraints - The primary database can not be stopped for any significant period of time, even for a maintenance window. The Secondary (read replica) similarly cannot be stopped for any period of time. A separate read replica was created to allow this work to be done and still maintain a strong Disaster Recovery Posture. 
+
+Database size: approximately 4.5 TB. 10s of billions of rows with single tables being upwards of 1 TB.
+
+### Justification
+
+This strategy poses near *zero* risk to current production operations. If the migration is successful, the required maintenance window will just be a database "cut over". We utilize native Postgresql utilities and AWS DMS on top of `pglogical` for additional functionality and visibility. 
+
+We use `pglogical` to enable a cross version migration.  This allows us to minimize the number of maintenance windows and high risk tasks within the production setup. 
+
+### Naming Conventions
+
+Primary = Source = Prouduction Database
+Secondary = Production Read Replica
+Secondary = Maintenance Read Replica
+RDS = New Primary = Target 
+
+In the steps below, I will use a combination of the above naming conventions.  When discussing replication via DMS CDC leveraging pglogical, I will simply say `source` or `target`. In a migration attempt, the source for the CDC portion is the Production Database, the primary. The `target` will always refer to RDS.
+
+For the Full Load portion, the Maintenance Read replica will be utilized and I will refer to it simply as "the secondary" or the "maintenance read replica".  The Production Read Replica is never touched during this process.
+
+# Prerequisite Steps
+
+Install the `pglogical` extension on the Source (production primary database), and on the Target (RDS). The secondary does not require the `pglogical` extension.
+
+### Create a Maintenance Read Replica
+
+In order to minimize risk and in an effort to not interfere with production, a fresh, separate read replica should be created and utilized for the migration. Replication will be paused on this separate read replica before performing full load, the current production read replica will continue to replicate and can still be used in the case of an emergency. Information surrounding the point in time backup will be garnered from this read replica and used to load any changed data after full load is completed.
```

Comment:
"...will be garnered from this NEW read replica and used..."  - just to clarify, you mention both read replicas so its a little muddled. actually im not entirely sure which it should be referring to